build.proj

```diff
@@ -0,0 +1,107 @@
+<Project DefaultTargets="Pack">
+	<PropertyGroup>
+		<Version Condition=" '$(Version)' == '' ">99.0-pre</Version>
+		<TargetFrameworks Condition=" '$(TargetFrameworks)' == '' ">netstandard2.1;xamarin.mac20;xamarin.ios10</TargetFrameworks>
+		<Configuration Condition=" '$(Configuration)' == '' ">Release</Configuration>
+
+		<!-- Managed code -->
+		<SrcPath Condition=" '$(SrcPath)' == '' ">$(MSBuildThisFileDirectory)src\</SrcPath>
+		<BaseOutputPath Condition=" '$(BaseOutputPath)' == '' ">$(MSBuildThisFileDirectory)managed\</BaseOutputPath>
+		<OutputPath Condition=" '$(OutputPath)' == '' ">$(BaseOutputPath)$(Configuration)\</OutputPath>
+
+		<!-- Native code -->
+		<NativeBaseOutputPath Condition=" '$(NativeBaseOutputPath)' == '' ">$(MSBuildThisFileDirectory)native\</NativeBaseOutputPath>
+		<NativeOutputPath Condition=" '$(NativeOutputPath)' == '' ">$(NativeBaseOutputPath)$(Configuration)\</NativeOutputPath>
+
+		<!-- SwiftUIGlue -->
+		<SwiftUIGlueDir Condition=" '$(SwiftUIGlueDir)' == '' ">$(SrcPath)SwiftUIGlue\</SwiftUIGlueDir>
+		<SwiftUIGlueDylib Condition=" '$(SwiftUIGlueDylib)' == '' ">$(SwiftUIGlueDir).build\$(Configuration.ToLowerInvariant())\libSwiftUIGlue.dylib</SwiftUIGlueDylib>
+		<SwiftUIGlueXcodeproj Condition=" '$(SwiftUIGlueXcodeproj)' == '' ">$(SwiftUIGlueDir)SwiftUIGlue.xcodeproj</SwiftUIGlueXcodeproj>
+	</PropertyGroup>
+
+	<ItemGroup>
+		<SwiftUIGlueTargets Condition="$(TargetFrameworks.Contains('xamarin.mac20'))" Include="SwiftUIGlueMac" />
+		<SwiftUIGlueTargets Condition="$(TargetFrameworks.Contains('xamarin.ios10'))" Include="SwiftUIGlueXcode" />
+
+		<SwiftUIGlueSources Include="$(SwiftUIGlueDir)*.swift" />
+
+		<!-- Mac -->
+		<!-- FIXME: Include dSYM? -->
+		<SwiftUIGlueDylibOutputs Include="$(SwiftUIGlueDylib)" />
+		<SwiftUIGlueMacOutputs Include="@(SwiftUIGlueDylibOutputs->'$(NativeOutputPath)xamarin.mac20\%(RecursiveDir)%(Filename)%(Extension)')" />
+
+		<!-- All other platforms -->
+		<SwiftUIGlueSdks Condition="$(TargetFrameworks.Contains('xamarin.ios10'))" Include="iphoneos;iphonesimulator" TargetFramework="xamarin.ios10" />
+
+		<SwiftUIGlueSdks Update="@(SwiftUIGlueSdks)">
+			<IntermediateFrameworkDir>$(SwiftUIGlueDir)build\$(Configuration)-%(Identity)\SwiftUIGlue.framework\</IntermediateFrameworkDir>
+			<OutputFrameworkDir>$(NativeOutputPath)%(TargetFramework)\SwiftUIGlue.framework\</OutputFrameworkDir>
+		</SwiftUIGlueSdks>
+	</ItemGroup>
+
+	<Target Name="SwiftUIGlueXcode">
+		<Exec
+			Condition="!Exists('$(SwiftUIGlueXcodeproj)')"
+			Command="swift package generate-xcodeproj"
+			WorkingDirectory="$(SwiftUIGlueDir)" />
+
+		<Exec
+			Command="xcodebuild -project $(SwiftUIGlueXcodeproj) -configuration $(Configuration) -sdk %(SwiftUIGlueSdks.Identity)"
```

Comment:
@chkn looks like you need to check in `project.pbxproj`, because when this command runs, I get the following error....

`/Users/dominique/Development/xamarin/Xamarin.SwiftUI/src/SwiftUIGlue/SwiftUIGlue.xcodeproj cannot be opened because it is missing its project.pbxproj`

Reply:
That should've been generated by the previous command. Though now I realize that it's possible for that file to be deleted if you run `git clean` without passing -d. So I modified the condition on that command to regenerate it if `project.pbxproj` is missing. Good catch!